testapp/src/main/java/com/rakuten/tech/mobile/testapp/analytics/rat_wrapper/IRatComponent.kt

```diff
@@ -11,4 +11,22 @@ interface IRatComponent {
      * Set Custom rat events
      */
     fun setCustomRatEvent(ratEvent: RATEvent)
```

Comment:
It seems like there is currently a lot of duplicate code between classes and as I was saying above it seems kind of overly complex to have 3 different interfaces. It also seems a little complicated that you need to both set the RatEvent with `setCustomRatEvent`. Could you just create and send the event in the same step?

Maybe you could instead have this single `RatComponent` interface which just defines the variables that need to be set. For example:

```
interface RatComponent {
    val pageName: String
    val siteSection: String
}

abstract class RatActivity : AppCompatActivity(), RatComponent {

    override fun onResume() {
        DemoAppAnalytics.init(AppSettings.instance.projectId).sendAnalytics(RATEvent(
            event = EventType.PAGE_LOAD,
            pageName = pageName,
            siteSection = siteSection
        ))

        super.onResume()
    }
}

class MyActivity : RatActivity() {
    override val pageName = "MyPageName"
    override val siteSection= "MySiteSection"
}
```

Reply:
@corycaywood actually `setCustomRatEvent ` is a extra method if user want to change the values of RAT events, if user does not set it, it will automatically send the default event..